scipy/sparse/linalg/eigen/lobpcg/lobpcg.py

```diff
@@ -31,9 +31,20 @@ def save(ar, fileName):
     savetxt(fileName, ar, precision=8)
 
 
-def _assert_symmetric(M, rtol=1e-5, atol=1e-8):
-    assert_allclose(M.T.conj(), M, rtol=rtol, atol=atol)
+def _report_nonhermitian(M, a, b, name):
+    """
+    Report if `M` is not a hermitian matrix given the tolerances `a`, `b`.
+    """
+    from scipy.linalg import norm
+
+    md = M - M.T.conj()
 
+    nmd = norm(md, 1)
+    tol = max(np.spacing(10**a), (10**b)*norm(M, 1))
```

Comment:
I made a mistake here this line should probably be

```python
tol = np.spacing(max(10**a, (10**b)*norm(M, 1)))
```

I am typing on the phone so I can't see all at once but the idea is simply to take the maximum of `1000` and `some scaling * norm of the matrix` and take the np.spacing of it. 

However I don't know what does print statements do. Are you supposed to give a warning there?

Reply:
I will fix `tol` (if this function survives, see below), thanks!

It is not really a warning, this function is only meant to print info on potential non-symmetry when the verbosity level is non-zero. I have added it instead of the commented-out assertions. I am not sure about its usefulness, though, @lobpcg?

Reply:
In principle, it is useful to cheaply check the symmetry, especially since the matrices of the eigenproblem to solve can be passed as user-defined functions... 

It is probably best to use one of the already existing functions, such as
https://docs.scipy.org/doc/numpy/reference/generated/numpy.testing.assert_allclose.html#numpy.testing.assert_allclose 

The specific values of rtol and atol are tricky to choose well, since they depend on the data type, float32 vs foat64, the matrix size, and the algorithm used for the comparison... However, it is not so practically important now, since we currently put the check only for a non-default verbosityLevel. 

@rc whatever changes you finally make here, please remember to sync with https://github.com/lobpcg/scikit-learn/blob/lobpcg_svd/sklearn/utils/lobpcg.py and check that it does not lead to formatting errors.

Reply:
Yes, choosing rtol and atol is tricky, that is why @ilayn proposed the way above.